lom/src/engine/sequenceHandler.go

```diff
@@ -0,0 +1,832 @@
+        if data.ResultCode != 0 {
+            /*
+             * No point in creating sequence. Call Raise request, which will succeed
+             * if first action in sequence.
+             *
+             * NOTE: An action may go crazy returning too frequently. This is guarded
+             * as below.
+             * 1. A plugin's conf has min interval between two callbacks for same anomaly key.
+             * 2. PluginMgr has a moving window of last 100 instances for a key of action+anomalyKey
+             *    If that crosses an average of N seconds between 2, it disables the action and
+             *    de-register it.
+             *
+             * If first action, re-publish is not done, as detection has failed.
+             */
+            errCode = LoMResponseCode(data.ResultCode)
+            errStr = data.ResultStr
+            return
+        }
+        bs, err := GetConfigMgr().GetSequence(data.Action)
+        if err != nil {
+            /* No sequence for this action. Likely stale non-first action or config changed */
+            errCode = LoMMissingSequence
+            errStr = "No sequence found"
+            return
+        }
+        tnow := time.Now().Unix()
+        ctx := make([]*ActionResponseData, 0, len(bs.Actions))
+        ctx = append(ctx, data)
+
+        seq = &sequenceState_t {
+            anomalyInstanceId: anomalyID,
+            sequence: bs,
+            seqStartEpoch: tnow,
+            seqExpEpoch: tnow + int64(bs.Timeout),
+            ctIndex: 1,
+            context: ctx,
+        }
+        p.addSequence(seq, bs.Timeout)
+        return
+    }
+
+    /* reset current request */
+    seq.currentRequest = nil
+
+    /* Move ahead with next action */
+    seq.context = append(seq.context, data)
+    seq.ctIndex++
+
+    /* Save last result */
+    errCode = LoMResponseCode(data.ResultCode)
+    errStr = data.ResultStr
+
+    /* Process the sequence */
+    if data.ResultCode != 0 {
+        /* Need to abort the sequence */
+        return
+    }
+
+    if seq.ctIndex >= len(seq.sequence.Actions) {
+        /* WooHoo we are really complete  */
+        seq.sequenceStatus = sequenceStatus_complete
+        return
+    }
+
+    errCode, errStr, err = p.resumeSequence(seq)
+    if errCode == LoMActionActive {
+        /*
+         * May be a prior seq left behind an active request. We got to wait
+         * till it responds or this sequence timeout, whichever earlier.
+         */
+         errCode = LoMResponseOk
+     }
+    return
+}
+
+/*
+ * resumeSequence
+ *
+ * Validate ctIndex. Attempt to raise request per ctIndex. If failed, return
+ * appropriate error. If succeeded, add it to corresponding client's pending
+ * server requests and list of active requests. Set it as current request of
+ * this sequence.
+ */
+func (p *SeqHandler_t) resumeSequence(seq *sequenceState_t) (errCode LoMResponseCode, errStr string, err error) {
+    regF := GetRegistrations()
```

Comment:
registrations ?

Reply:
All active registrations, clients & actions.

Reply:
I meant renaming regF to registrations.

Reply:
It is just local var. So as long as it implies, good enough. Too long each line in code longer and hence harder to read.